brainzutils/timeseries_stats.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import division
+
+import datetime
+
+import six
+
+from brainzutils import cache
+
+NAMESPACE_STATS = "timeseries_stats"
+
+# Keep this many hours of old cache items in each key
+HOURS_TO_KEEP = 12
+
+STATS_RANGE_MINUTE = 1
+STATS_RANGE_10MIN = 10
+STATS_RANGE_HOUR = 60
+
+
+class TimeseriesStats:
```

Comment:
I'm not sure if making an object for each metric is ideal.

I would prefer if I was able to do something like

```
TimeseriesStats.increment('my_stat_name', time_range)
```

Also, would prefer to name this Metrics maybe.

Reply:
yeah, I spent a bit of time thinking about this. I didn't want to have to pass in the time range every single time that I called it. Imagine that this could be an app-level constant that you can import and call. We could have a version that has a default resolution.

Metrics is a good name, I took a long time coming up with this and I'm still not happy with it.

You're right that it could be as simple as calling increment_something('stat_name') - I wasn't sure if the grouping by time range was important, or if granfa is able to group by any time range anyway. I was thinking about data storage - if we store a timestamp for every record, that's a new row every time an event happens. In the case that we have a counter that increments thousands to millions of times a day, how does this impact our storage? Is it more efficient to put it into 1 hour buckets if we don't need any more resolution? I need to follow this up with Zas. 

Would there be a case where we would have more than 1 increment in a second? if that was the case, how should we handle that? miliseconds in the timestamp? or just allow a maximum of one per second?